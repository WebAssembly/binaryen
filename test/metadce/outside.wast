(module
 (import "env" "js_func" (func $a_js_func))
 (import "env" "js_func_unused" (func $an_unused_js_func))
 (import "env" "DYNAMICTOP_PTR" (global $DYNAMICTOP_PTR$asm2wasm$import i32))
 (import "env" "DYNAMICTOP_PTR_unused" (global $DYNAMICTOP_PTR$asm2wasm$import_unused i32))
 (import "env" "memory" (memory $0 256 256))
 (import "env" "table" (table 10 10 funcref))

 (export "wasm_func" (func $a_wasm_func))
 (export "wasm_func_unused" (func $an_unused_wasm_func))

 (global $__THREW__ (mut i32) (i32.const 0))
 (global $__THREW__unused (mut i32) (i32.const 0))
 (global $from_segment (mut i32) (i32.const 0))
 (global $from_segment_2 (mut i32) (i32.const 0))
 (global $from_segment_never_used (mut i32) (i32.const 0))

 (data (i32.const 1024) "abcd")
 (data (global.get $from_segment) "abcd")
 (elem (global.get $from_segment_2) $table_func)

 (func $a_wasm_func
  (call $a_js_func)
  (drop (global.get $DYNAMICTOP_PTR$asm2wasm$import))
  (drop (global.get $__THREW__))
 )
 (func $an_unused_wasm_func
  (drop (global.get $DYNAMICTOP_PTR$asm2wasm$import_unused))
  (drop (global.get $__THREW__unused))
 )
 (func $table_func
 )
)

