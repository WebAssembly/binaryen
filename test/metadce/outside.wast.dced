(module
 (type $none_=>_none (func))
 (import "env" "memory" (memory $0 256 256))
 (data (i32.const 1024) "abcd")
 (data (global.get $from_segment) "abcd")
 (import "env" "table" (table $0 10 10 funcref))
 (elem (global.get $from_segment_2) $table_func)
 (import "env" "DYNAMICTOP_PTR" (global $DYNAMICTOP_PTR$asm2wasm$import i32))
 (import "env" "js_func" (func $a_js_func))
 (global $__THREW__ (mut i32) (i32.const 0))
 (global $from_segment (mut i32) (i32.const 0))
 (global $from_segment_2 (mut i32) (i32.const 0))
 (export "wasm_func" (func $a_wasm_func))
 (func $table_func (; 1 ;)
  (nop)
 )
 (func $a_wasm_func (; 2 ;)
  (call $a_js_func)
  (drop
   (global.get $DYNAMICTOP_PTR$asm2wasm$import)
  )
  (drop
   (global.get $__THREW__)
  )
 )
)
