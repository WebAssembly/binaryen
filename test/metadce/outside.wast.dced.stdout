unused: export$wasm_func_unused$12
unused: func$an_unused_wasm_func$1
unused: global$__THREW__unused$4
unused: global$from_segment$5
unused: global$from_segment_2$6
unused: global$from_segment_never_used$7
